pym/portage/meta-manifest.py

```diff
@@ -0,0 +1,185 @@
+        InvalidDataType, MissingParameter, PermissionDenied, \
+        PortageException, PortagePackageException
+from const import (MANIFEST1_HASH_FUNCTIONS, MANIFEST2_HASH_DEFAULTS,
+        MANIFEST2_HASH_FUNCTIONS, MANIFEST2_IDENTIFIERS, MANIFEST2_REQUIRED_HASH)
+from portage.localization import _
+
+if sys.hexversion >= 0x3000000:
+	# pylint: disable=W0622
+	_unicode = str
+	basestring = str
+else:
+	_unicode = unicode
+
+def guessManifestFileType(filename):
+        """ Perform a best effort guess of which type the given filename is, avoid using this if possible """
+        if filename.startswith("files" + os.sep + "digest-"):
+                return None
+        elif filename.startswith("files" + os.sep):
+                return "AUX"
+        elif filename.endswith(".ebuild"):
+                return "EBUILD"
+        elif filename == "Manifest" or filename == "MetaManifest":
+                return "MANIFEST"
+        elif filename.endswith(".eclass"):
+                return "ECLASS"
+        elif filename in ["ChangeLog", "metadata.xml"]:
+                return "MISC"
+        elif filename.endswith(".sh"):
+	        return "EXEC"
+        else:
+                return "OTHER"
+
+class MetaManifest(manifest.Manifest):
+
+	def getFullname(self):
+                """ Returns the absolute path to the Manifest file for this instance """
+                return os.path.join(self.pkgdir, "MetaManifest")
+
+	def sign(self):
+		'''Signs MetaManifest file with the default PORTAGE_GPG_KEY''' 
+		filename = self.getFullname()
+		portage_settings = portage.config(clone=portage.settings)
+		portage_portdbapi = portage.portdbapi(portage_settings)
+		gpgcmd = portage_settings.get("PORTAGE_GPG_SIGNING_COMMAND")
+		if gpgcmd in [None, '']:
+			raise portage.exception.MissingParameter(
+				"PORTAGE_GPG_SIGNING_COMMAND is unset! Is make.globals missing?")
+		if "${PORTAGE_GPG_KEY}" in gpgcmd and "PORTAGE_GPG_KEY" not in portage_settings:
+			raise portage.exception.MissingParameter("PORTAGE_GPG_KEY is unset!")
+		if "${PORTAGE_GPG_DIR}" in gpgcmd:
+			if "PORTAGE_GPG_DIR" not in portage_settings:
+				portage_settings["PORTAGE_GPG_DIR"] = os.path.expanduser("~/.gnupg")
+			else:
+				portage_settings["PORTAGE_GPG_DIR"] = os.path.expanduser(portage_settings["PORTAGE_GPG_DIR"])
+			if not os.access(portage_settings["PORTAGE_GPG_DIR"], os.X_OK):
+				raise portage.exception.InvalidLocation(
+					"Unable to access directory: PORTAGE_GPG_DIR='%s'" %
+					portage_settings["PORTAGE_GPG_DIR"])
+		gpgvars = {"FILE": filename}
+		for k in ("PORTAGE_GPG_DIR", "PORTAGE_GPG_KEY"):
+			v = portage_settings.get(k)
+			if v is not None:
+				gpgvars[k] = v
+		gpgcmd = portage.util.varexpand(gpgcmd, mydict=gpgvars)
+		gpgcmd = portage.util.shlex_split(gpgcmd)
+		gpgcmd = [portage._unicode_encode(arg, encoding=portage._encodings['fs'], errors='strict') for arg in gpgcmd]
+		return_code = subprocess.call(gpgcmd)
```

Comment:
This is a minor nitpick as I have no idea whether this would be preferred or not in portage code base but what about using [`subprocess.check_call`](https://docs.python.org/2/library/subprocess.html#subprocess.check_call)